client/components/sites-dropdown/test/index.js

```diff
@@ -1,48 +1,64 @@
 /**
  * @jest-environment jsdom
  */
-import { shallow } from 'enzyme';
+import userEvent from '@testing-library/user-event';
+import preferences from 'calypso/state/preferences/reducer';
+import ui from 'calypso/state/ui/reducer';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
 import { SitesDropdown } from '..';
 
 const noop = () => {};
 
+const render = ( el, options ) =>
+	renderWithProvider( el, {
+		...options,
+		reducers: { ui, preferences },
+		initialState: { preferences: { remoteValues: {} } },
+	} );
+
 describe( 'index', () => {
 	describe( 'component rendering', () => {
+		const props = {
+			selectedSiteId: 12345678,
+			primarySiteId: 12345678,
+		};
 		test( 'should render a dropdown component initially closed', () => {
-			const sitesDropdown = shallow( <SitesDropdown /> );
-			expect( sitesDropdown.hasClass( 'sites-dropdown' ) ).toEqual( true );
-			expect( sitesDropdown.hasClass( 'is-open' ) ).toEqual( false );
+			const { container } = render( <SitesDropdown { ...props } /> );
+
+			expect( container.firstChild ).toHaveClass( 'sites-dropdown' );
+			expect( container.firstChild ).not.toHaveClass( 'is-open' );
 		} );
 
-		test( 'with multiple sites, should toggle the dropdown when it is clicked', () => {
+		test( 'with multiple sites, should toggle the dropdown when it is clicked', async () => {
+			const user = userEvent.setup();
 			const toggleOpenSpy = jest.spyOn( SitesDropdown.prototype, 'toggleOpen' );
-			const sitesDropdown = shallow( <SitesDropdown hasMultipleSites={ true } /> );
 
-			sitesDropdown.find( '.sites-dropdown__selected' ).simulate( 'click' );
+			const { container } = render( <SitesDropdown { ...props } hasMultipleSites /> );
+
+			await user.click( container.querySelector( '.sites-dropdown__selected' ) );
```

Comment:
I'd love to see this component to be more accessible.

Reply:
Indeed!